### PR TITLE
Climb commands

### PR DESCRIPTION
**Pull request by josiahcrossman**
Made auto climb system, added power draw methods for various subsystems.

### CODE REVIEW

File: src/main/java/frc/robot/commands/climb/ClimbAutomation.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    clampPistRight = climbStates.UNKNOWN; 
+    climbSystem = climbStates.READY; 
+  }
+
+  // Called every time the scheduler runs while the command is scheduled.
+  @Override
+  public void execute() {
+    if (prioritizedClimbButton.get()){
+
+
```
Comment (Xmde) [inline src/main/java/frc/robot/commands/climb/ClimbAutomation.java:261]: I feel like you are missing stuff here. Are you still planning to implement this or did you forget to commit/push your changes?

File: src/main/java/frc/robot/commands/shooter/SemiAutoShooterAssembly.java [hunk 2]
```diff
@@ -0,0 +1,180 @@
+package frc.robot.commands.shooter;
```
Comment (Xmde) [inline src/main/java/frc/robot/commands/shooter/SemiAutoShooterAssembly.java:1]: See the previous comment.

File: src/main/java/frc/robot/commands/shooter/FullAutoShooterAssembly.java [hunk 3]
```diff
@@ -0,0 +1,196 @@
+package frc.robot.commands.shooter;
```
Comment (Xmde) [inline src/main/java/frc/robot/commands/shooter/FullAutoShooterAssembly.java:1]: I feel like you accidentally merged in the Shooter Assembly branch or made this branch from that one. Clean up the branch and remove the files associated with the shooter assembly.

File: src/main/java/frc/robot/commands/climb/ClimbAutomation.java [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+  }
+
+  // Called when the command is initially scheduled.
+  @Override
+  public void initialize() {
+    climb.stop();
+    prioritizedClimbButton =  climbButton.prioritize(AxisPriorities.DEFAULT);
+    motorLeft = climbStates.IDLE;
+    motorRight = climbStates.IDLE; 
+    pivotPistLeft = climbStates.UNKNOWN;
```
Comment (Xmde) [inline src/main/java/frc/robot/commands/climb/ClimbAutomation.java]: These states are known to be retracted at the beginning of the match.

**Review by Xmde — CHANGES_REQUESTED** [review]
Looks good but I feel like you are missing the automation in the ClimbAutomation command. You also have the Shooter automation code in this branch and that should get cleaned up and fixed.

File: src/main/java/frc/robot/subsystems/Climb.java [hunk 5]
```diff
@@ -32,8 +32,10 @@
             new CANSparkMax(Ports.RIGHT_CLIMB_MOTOR, CANSparkMaxLowLevel.MotorType.kBrushless);
 
     // Pivot Solenoid
-    private final Solenoid pivot =
-            new Solenoid(Ports.PCM_1, PneumaticsModuleType.REVPH, Ports.PIVOT_CLIMB_SOLENOID);
+    private final Solenoid pivotLeft =
```
Comment (django-wardlow) [inline src/main/java/frc/robot/subsystems/Climb.java]: pivot left and right are on the same solenoid

File: src/main/java/frc/robot/commands/climb/ClimbMO.java [hunk 6]
```diff
@@ -19,9 +18,13 @@
     OIAxis controlAxis;
     OIAxis.PrioritizedAxis prioritizedControlAxis;
 
-    // button for the pivot solenoid
-    OIButton pivotButton;
-    OIButton.PrioritizedButton prioritizedPivotButton;
+    // button for the left pivot solenoid
+    OIButton leftPivotButton;
```
Comment (django-wardlow) [inline src/main/java/frc/robot/commands/climb/ClimbMO.java]: left and right pivot are the same, they can not be controled independently

File: src/main/java/frc/robot/commands/climb/ClimbAutomation.java [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+      READY
+    
+    }
+  
+  // Creating different Climb states for tracking processes
+    //motors
+    climbStates motorLeft;
+    climbStates motorRight;
+    // pivot pistons
+    climbStates pivotPistLeft;
```
Comment (django-wardlow) [inline src/main/java/frc/robot/commands/climb/ClimbAutomation.java]: left and right pivots are on the same solinoid

**Review by django-wardlow — CHANGES_REQUESTED** [review]

**Comment by Xmde** [comment]
Is this still a pull request you want me to take a look at or is all of this code already in the testing branch?

**Comment by Xmde** [comment]
Closing PR because changes already in testing branch.

**Pull request closed without merging**